### ISSUE TITLE
Vectorise Dirichlet distribution

### ISSUE THREAD

**Issue by andrjohns** [stan-dev/math#1362]
## Description
[Requested by a user on the forums](https://discourse.mc-stan.org/t/reparameterization-example-in-manual/3199)

Add broadcasting to the dirichlet distribution, so that the following are valid:
```
std::vector<VectorXd> ~ dirichlet(VectorXd)
std::vector<VectorXd> ~ dirichlet(std::vector<VectorXd>)
```

This should simplify the declaration for users, so that there's no need to loop over each individual vector in the array

#### Current Version:
v2.20.0

**Comment by bob-carpenter** [comment]
For completeleness, we also need

Matrix<T, -1, 1> ~ dirichlet(vector<Matrix<T, -1, 1>)

**Comment by mcol** [comment]
Fixed by #1362.
